next.config.js

```diff
@@ -16,6 +20,44 @@ const temporary_hideDocsPaths = {
   has: [{ type: 'host', value: '(^(?!.*(waypointproject)).*$)' }],
 }
 
+/**
+ * Reads in config files from config/[env].json and replaces references in the code
+ * with the literal values using webpack.DefinePlugin
+ */
+function HashiConfigPlugin() {
+  const env = process.env.HASHI_ENV || 'development'
+  const envConfigPath = path.join(process.cwd(), 'config', `${env}.json`)
+
+  function getHashiConfig(path) {
+    try {
+      const envConfig = JSON.parse(fs.readFileSync(path))
+      const ret = flat(envConfig, {
+        safe: true,
+      })
+      return ret
+    } catch (err) {
+      console.log('Error loading environment config:', err)
+      return {}
+    }
+  }
+
+  return new webpack.DefinePlugin({
```

Comment:
I'm opting to use `DefinePlugin` for the first iteration because this injects the literal value into the code, which enables dead code elimination.

So if we use this for flags:
```js
if (config.flags.enable_new_downloads_page) {
  return <NewDownloadsPage />
} else {
  return <OldDownloadsPage />
}
```

the value would get replaced and the dead code path would get removed:
```js
if (true) {
  return <NewDownloadsPage />
} else {
  // this code path is removed from the bundle
  return <OldDownloadsPage />
}
```